src/main/java/org/opentripplanner/netex/loader/mapping/FlexStopLocationMapper.java

```diff
@@ -0,0 +1,81 @@
+package org.opentripplanner.netex.loader.mapping;
+
+import net.opengis.gml._3.AbstractRingPropertyType;
+import net.opengis.gml._3.LinearRingType;
+import net.opengis.gml._3.PolygonType;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.CoordinateSequence;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.LinearRing;
+import org.locationtech.jts.geom.Polygon;
+import org.opentripplanner.api.resource.CoordinateArrayListSequence;
+import org.opentripplanner.common.geometry.GeometryUtils;
+import org.opentripplanner.model.FlexStopLocation;
+import org.rutebanken.netex.model.FlexibleArea;
+import org.rutebanken.netex.model.FlexibleStopPlace;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FlexStopLocationMapper {
+
+  private final FeedScopedIdFactory idFactory;
+
+  public FlexStopLocationMapper(FeedScopedIdFactory idFactory) {
+    this.idFactory = idFactory;
+  }
+
+  /**
+   * Maps NeTEx FlexibleStopPlace to FlexStopLocation. This currently does not support
+   * FlexLocationGroup, as an equivalent is not defined in the NeTEx Nordic profile.
+   */
+  public FlexStopLocation map(FlexibleStopPlace flexibleStopPlace) {
+    FlexStopLocation result = new FlexStopLocation(idFactory.createId(flexibleStopPlace.getId()));
+    result.setName(flexibleStopPlace.getName().getValue());
+
+    Object flexibleAreaOrFlexibleAreaRefOrHailAndRideArea = flexibleStopPlace
+        .getAreas()
+        .getFlexibleAreaOrFlexibleAreaRefOrHailAndRideArea()
+        .get(0); // Only one area allowed in NeTEx Nordic profile.
```

Comment:
Wouldn't it be safe to always construct a multipolygon?

Reply:
We could construct a multipolygon here, but it would only contain one polygon, as we are mapping from OpenGIS polygon in the NeTEx XML.